app/permissions/experiment.rb

```diff
@@ -0,0 +1,192 @@
+  end
+
+  # module: complete/uncomplete
+  can :complete_module do |user, my_module|
+    user.is_technician_or_higher_of_project?(my_module.experiment.project)
+  end
+
+  # TODO: When rebasing on top of refactored projects permissions, just call
+  # can_create_comment_in_project?(user, my_module.experiment.project) instead
+  # module: create comment
+  # result: create comment
+  # step: create comment
+  can :create_comment_in_module do |user, my_module|
+    user.is_technician_or_higher_of_project?(my_module.experiment.project)
+  end
+
+  %i(manage_module
+     manage_users_in_module
+     delete_or_archive_result
+     assign_sample_to_module
+     complete_module
+     create_comment_in_module).each do |perm|
+    can perm do |_, my_module|
+      my_module.active? &&
+        my_module.experiment.active? &&
+        my_module.experiment.project.active?
+    end
+  end
+end
+
+Canaid::Permissions.register_for(Protocol) do
+  # protocol in module: read
+  # step: read (read comments, read assets), download assets
+  can :read_protocol_in_module do |user, protocol|
+    if protocol.in_module?
+      can_read_experiment?(user, protocol.my_module.experiment)
+    else
+      false
+    end
+  end
+
+  # protocol in module: create, update, delete, unlink, revert, update from
+  # protocol in repository, update from file
+  # step: create, update, delete, reorder
+  can :manage_protocol_in_module do |user, protocol|
+    if protocol.in_module?
+      can_manage_module?(user, protocol.my_module)
+    else
+      false
+    end
+  end
+
+  # step: complete/uncomplete
+  can :complete_or_checkbox_step do |user, protocol|
+    if protocol.in_module?
+      can_complete_module?(user, protocol.my_module)
+    else
+      # In repository, user cannot complete steps
+      false
+    end
+  end
+
+  %i(read_protocol_in_module
+     manage_protocol_in_module
+     complete_or_checkbox_step)
+    .each do |perm|
+    can perm do |_, protocol|
+      my_module = protocol.my_module
+      my_module.active? &&
+        my_module.experiment.active? &&
+        my_module.experiment.project.active?
+    end
+  end
+end
+
+Canaid::Permissions.register_for(Comment) do
+  # module: update/delete comment
+  # result: update/delete comment
+  # step: update/delete comment
+  can :manage_comment_in_module do |user, comment|
+    project = case comment.is_a?
+              when TaskComment
+                comment.my_module.experiment.project
+              when ResultComment
+                comment.result.my_module.experiment.project
+              when StepComment
+                comment.step.protocol.my_module.experiment.project
+              end
+    project.present? &&
+      # TODO: When rebasing on top of refactored projects permissions, just call
+      # can_manage_comment_in_project?(user, project) instead
+      (user.is_owner_of_project(project) || comment.user == current_user)
+  end
+
+  %i(comment).each do |perm|
```

Comment:
why?

Reply:
because the permission is `manage_comment_in_module` :).